pilot/pkg/model/policyattachment.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"istio.io/api/type/v1beta1"
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/schema/gvk"
+)
+
+type policyTargetGetter interface {
+	GetTargetRef() *v1beta1.PolicyTargetReference
+	GetSelector() *v1beta1.WorkloadSelector
+}
+
+type policyMatch string
+
+const (
+	policyMatchSelector policyMatch = "selector"
+	policyMatchDirect   policyMatch = "direct"
+	policyMatchIgnore   policyMatch = "ignore"
+)
+
+func getPolicyMatcher(policyName string, opts workloadSelectionOpts, policy policyTargetGetter) policyMatch {
+	gatewayName, isGatewayAPI := opts.workloadLabels[constants.IstioGatewayLabel]
```

Comment:
I canot see where IstioGatewayLabel is set

Reply:
Lol, there's already another const for it. I'll switch to the preexisting one